Main.c

```diff
@@ -0,0 +1,17 @@
+#include "CheckRange.h"
+#include <assert.h>
+
+int main()
+{
+int *range;
+
+int array[] = {3, 3, 5, 4, 10, 11, 12};
+int arraysize = sizeof(array) / sizeof(array[0]);
+  
+range = CheckRange(array,arraysize);
+
+assert(*(range+4) == 3);

```

Comment:
test case asserts are not intuitive. One has to check the production code to understand what the test cases are supposed to check. Usually test cases are self explanatory and indicate what output the production code should be returning